src/spyglass/lfp/v1/analytic_signal.py

```diff
@@ -0,0 +1,95 @@
+import os
+import datajoint as dj
+import numpy as np
+import pandas as pd
+from scipy.signal import hilbert
+import math
+from spyglass.utils.dj_helper_fn import fetch_nwb
+from spyglass.lfp.v1.lfp import LFP,LFPBand
+from spyglass.common.common_interval import IntervalList
+from spyglass.common.common_nwbfile import AnalysisNwbfile
+
+schema = dj.schema("lfp_v1_analytic_signal")
+
+@schema
+class AnalyticSignalParameters(dj.Manual):
+    definition = """
+    analytic_signal_params_name: varchar(100) # name of the parameter set for detecting analytic signals
+    ---
+    analytic_signal_params: blob # a dict of analytic-signal detection parameters(the method name, e.g., hilbert transform, and specific parameters if any)
+    """
+
+@schema
+class AnalyticSignalSelection(dj.Manual):
+    definition = """
+    # Table that combines the LFP data, and parameters for generating analytic signals
+    -> LFPBand
+    -> IntervalList
+    -> AnalyticSignalParameters
+    ---
+    """
+
+@schema
+class AnalyticSignal(dj.Computed):
+    definition = """
+    -> AnalyticSignalSelection
+    ---
+    -> AnalysisNwbfile # the name of the nwb file with the analytic sinal & time stamps data
+    analytic_signal_object_id: varchar(40)  # the NWB object ID for loading this object from the file
+    """
+    def make(self, key):
+        # First, get the filtered lfp data
+        FilteredBand = (LFPBand() & key).fetch_nwb()[0]['filtered_data']
```

Comment:
I believe [PEP 8](https://peps.python.org/pep-0008/#function-and-variable-names) suggests using lowercase with underscores for variable and function names instead of camel case (which is reserved for classes).